### DISCUSSION TITLE
Overwriting third party components.

### DISCUSSION THREAD

**Discussion by awais019** [tailwindlabs/tailwindcss#14286] [Help] [answered]
I have been curious from some time that what's the best or at least good enough approach to overwrite styles for third party components. Those components have certain classes attached and I have to overwrite those classes in-order to customize them. And I'm using `@apply` directive but from the docs and some other blogs @adamwathan has suggested that `@apply` should be avoided. So, what can I do in this situation.

PS: I have also tried using `theme` function to apply styles

**Comment by wongjn** [comment, marked as answer]
You could consider:
- Applying classes to the component elements, if the components have APIs to do so.
- Using `theme()` in CSS.
- Using utility classes with [arbitrary variants](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) to target the component elements.
- Using [variants from custom Tailwind plugin(s)](https://tailwindcss.com/docs/plugins#adding-variants) to target the component elements.

**Comment by awais019** [comment]
But I'm talking about overwriting styles for already defined classes by the components

**Reply by wongjn** [reply]
Sure, my suggestions still apply.

**Reply by awais019** [reply]
Thanks

**Comment by shivangi-2001** [comment]
@wongjn 
Hi, I followed couple of issues in GitHub repo regarding css applied on third party tools like fullcalender, swiper. Before it applied using like these 
<img width="644" alt="Screenshot 2025-03-02 at 12 54 29 AM" src="https://github.com/user-attachments/assets/2f36bb1d-a6fc-48a2-8c38-225efb87d4ec" />
Now have it does not showing any changes in react 19 + tailwind css v4.0
using the above image code directly giving error:
<img width="704" alt="Screenshot 2025-03-02 at 12 55 46 AM" src="https://github.com/user-attachments/assets/3fac81e9-c9c2-45a4-9ce8-b0591d812344" />
I followed https://tailwindcss.com/docs/adding-custom-styles#adding-component-classes guideline but
it doesn't shows any changes.
<img width="716" alt="Screenshot 2025-03-02 at 1 01 21 AM" src="https://github.com/user-attachments/assets/83296362-8da5-47f7-87b9-2a682bc6d0d9" />


Can you tell me what can do about that.

**Reply by wongjn** [reply]
Try having the third-party CSS inside the `@layer components` too.

**Reply by shivangi-2001** [reply]
I am not able to use bg-black or other utility classes, i added into @ layer components
<img width="622" alt="Screenshot 2025-03-02 at 11 04 17 AM" src="https://github.com/user-attachments/assets/e838ddb6-eb22-406e-a5f2-2b65b023e4d7" />

**Reply by wongjn** [reply]
Try having the third-party CSS inside the `@layer components` too:

```css
@import "path/to/third-party/styles.css" layer(components);

…

@layer components {
  …
}
```
